rules/common/src/main/kotlin/io/nlopez/compose/rules/ParameterNaming.kt

```diff
@@ -0,0 +1,222 @@
+// Copyright 2023 Nacho Lopez
+// SPDX-License-Identifier: Apache-2.0
+package io.nlopez.compose.rules
+
+import io.nlopez.compose.core.ComposeKtConfig
+import io.nlopez.compose.core.ComposeKtVisitor
+import io.nlopez.compose.core.Emitter
+import io.nlopez.compose.core.report
+import io.nlopez.compose.core.util.composableLambdaTypes
+import io.nlopez.compose.core.util.isLambda
+import org.jetbrains.kotlin.psi.KtFunction
+
+class ParameterNaming : ComposeKtVisitor {
+
+    override fun visitComposable(
+        function: KtFunction,
+        autoCorrect: Boolean,
+        emitter: Emitter,
+        config: ComposeKtConfig,
+    ) = with(config) {
+        // For lambda parameters only: if it starts with `on`, we want it to not be in past tense, to be all consistent.
+        // E.g. onClick, onTextChange, onValueChange, and a myriad of other examples in the compose foundation code.
+
+        val lambdaTypes = function.containingKtFile.composableLambdaTypes
+
+        val errors = function.valueParameters
+            .filter { it.typeReference?.isLambda(lambdaTypes) == true }
+            .filter {
+                // As per why not force lambdas to all start with `on`, we cannot really know when they are used for
+                // lazy initialization purposes -- and also don't want to be overly annoying.
+                it.name?.startsWith("on") == true
+            }
+            .filter { it.name?.isPastTense == true }
+
+        for (error in errors) {
+            emitter.report(error, LambdaParametersInPresentTense)
+        }
+    }
+
+    private val String.isPastTense: Boolean
+        get() = endsWith("ed") || IrregularVerbsInPastTense.any { endsWith(it) }
+
+    companion object {
+        // A list of common irregular verbs in english, excluding those where present tense == past tense,
+        // according to chatgpt. If you stumble upon one not here, feel free to send a PR to add it.
+        private val IrregularVerbsInPastTense by lazy {
```

Comment:
i had to do this a bit by hand, and remove the ones that were == to the present. I hope I haven't left any.